Document Mustache template support in Search Relevance Workbench (#12810)
opensearch-project/documentation-website: _search-plugins/search-relevance/search-configurations.md

:---  | :--- | :---
`name` | String | The name of the search configuration.
`description` | String | Description of the search configuration.

> `query` | Object | Defines the query in OpenSearch query DSL. Use the `%SearchText%` placeholder or [Mustache](https://mustache.github.io/) template variables (such as {% raw %}`{{queryText}}`{% endraw %}) to substitute query set values at runtime. Needs to be escaped. For more information, see [Using Mustache templates](#using-mustache-templates).

Reviewer: ```suggestion
`query` | Object | Defines the query in OpenSearch query DSL, provided as a JSON string with the inner quotation marks escaped (for example, `"query": "{\"query\":{\"multi_match\":{...}}}"`). Use the `%SearchText%` placeholder or [Mustache](https://mustache.github.io/) template variables (such as {% raw %}`{{queryText}}`{% endraw %}) to substitute query set values at runtime. For more information, see [Using Mustache templates](#using-mustache-templates).
```

}
```

### Using Mustache templates

Instead of the `%SearchText%` placeholder, a query can use [Mustache](https://mustache.github.io/) template variables. If the `query` contains `{{`, it is rendered as a Mustache template; otherwise, the `%SearchText%` placeholder is substituted. Existing `%SearchText%` configurations continue to work unchanged.

The following variables are available when a template is rendered:

> * `{{query_string}}`: The `queryText` of the current query set entry.

Reviewer 2: Can we make it to be same as field name from query set?

```suggestion
* `{{queryText}}`: The `queryText` of the current query set entry.
```

* Any custom field defined on the query set entry, referenced by its field name (for example, `{{category}}`). For more information, see [Query sets]({{site.url}}{{site.baseurl}}/search-plugins/search-relevance/query-sets/).

Substituted values are automatically JSON-escaped, so query text containing quotation marks or other special characters does not break the query.

Mustache partials (`{{>...}}`) are not supported and cause the query to be rejected.
{: .note}

#### Example request: Creating a search configuration that uses a Mustache template

The following request creates a search configuration that matches the user query against the `title` field and filters results by a `category` custom field provided by each query set entry:

```json
PUT _plugins/_search_relevance/search_configurations
{
  "name": "mustache_filter",

>   "query": "{\"query\":{\"bool\":{\"must\":[{\"match\":{\"title\":\"{{query_string}}\"}}],\"filter\":[{\"term\":{\"category\":\"{{category}}\"}}]}}}",

Reviewer 2: ```suggestion
  "query": "{\"query\":{\"bool\":{\"must\":[{\"match\":{\"title\":\"{{queryText}}\"}}],\"filter\":[{\"term\":{\"category\":\"{{category}}\"}}]}}}",
```
